Scoop/Controller/Ride Posting/PostRideTripDetailsViewController.swift

```diff
@@ -80,23 +83,23 @@ class PostRideTripDetailsViewController: OnboardingViewController {
             self.ride.departureDatetime = dateFormatter.string(from: date)
             self.ride.description = self.detailsTextField.text ?? ""
             
-            self.navigationController?.pushViewController(PostRideSummaryViewController(currentRide: self.ride), animated: true)
+            self.delegate?.didTapNext(navCtrl, nextViewController: nil)
         }
```

Comment:
Not sure how you implemented it but be careful not to cause any retain cycles since you have a strong reference to `self` in this completion handler. If it's a property of your class and you're passing it to other classes then yes, make it a weak reference so that it can be dereferenced. Otherwise, I believe you're fine if it's within this class. I would check to see if your view controllers are being dereferenced when dismissed. Had to deal with retain cycles in one of the apps I made one time and it was painful.